spring-data-relational/src/main/java/org/springframework/data/relational/core/sqlgeneration/SingleQuerySqlGenerator.java

```diff
@@ -238,9 +242,10 @@ private String getIdentifierProperty(List<AggregatePath> paths) {
 
 	private static AnalyticFunction createRowNumberExpression(AggregatePath basePath, Table table,
 			String rowNumberAlias) {
+		AggregatePath.ColumnInfos reverseColumnInfos = basePath.getTableInfo().reverseColumnInfos();
 		return AnalyticFunction.create("row_number") //
-				.partitionBy(table.column(basePath.getTableInfo().reverseColumnInfo().name())) //
-				.orderBy(table.column(basePath.getTableInfo().reverseColumnInfo().name())) //
+				.partitionBy(reverseColumnInfos.toList(ci -> table.column(ci.name()))) //
```

Comment:
Duplicate code with `ColumnInfos` creating a list of columns.

Reply:
Fixed